spec/lib/file_tree_spec.rb

```diff
@@ -179,8 +179,22 @@
       expect(js_tree).to be_a(String)
     end
 
-    it 'produces the required JSON format' do
-      expect(JSON.parse(js_tree).deep_symbolize_keys).to eq(core: {data: file_tree.send(:map_to_js_tree, file_tree)})
+    context 'without any file' do
+      it 'produces the required JSON format' do
+        expect(JSON.parse(js_tree).deep_symbolize_keys).to eq(core: {data: []})
+      end
+    end
+
+    context 'with files' do
+      let(:files) { FactoryBot.build_list(:file, 10, context: nil, path: 'foo/bar/baz') }
+      let(:file_tree) { described_class.new(files) }
+      let(:js_tree) { file_tree.to_js_tree }
+
+      it 'produces the required JSON format with a file' do
+        # We ignore the root node and hence remove it here
```

Comment:
I don't quite see any removal here.

That said, several tests in this file are very implementation-specific, where I would expect them to only test the behavior of `FileTree` that's observable from the outside.

(Not exactly a new problem, so not a blocker, but I wanted to bring it up.)

Reply:
Both points are valid. I changed the comment to reflect that nothing is removed (in my assumption and compared to the old approach, the root node is _ignored_ and not considered for the comparison).

Regarding the test cases in this file: Indeed, many of the tests are implement-specific and not very general. As long as they work (and the implementation of the `FileTree` is not changed, I didn't consider changing it. Whenever it breaks, I would rather adopt them instead of tediously fixing them.

Reply:
Thanks.
When the implementation doesn't change, unit tests are basically no-ops anyways. 🤣 

No rush, and certainly no blocker, it just stood out to me.